examples/amplify-react/src/master.js

```diff
@@ -0,0 +1,240 @@
+import { Role } from 'amazon-kinesis-video-streams-webrtc'
+import WebRtcClient from './webRtcClient'
+
+export default class Master {
+  constructor(setState, onStatsReport, onRemoteDataMessage) {
+    this.setState = setState
+    this.onStatsReport = onStatsReport
+    this.onRemoteDataMessage = onRemoteDataMessage
+    this.signalingClient = null
+    this.peerConnectionByClientId = {}
+    this.dataChannelByClientId = {}
+    this.localStream = null
+    this.remoteStreams = []
+    this.peerConnectionStatsInterval = null
+    this.remoteViewSrcObject = null
+    this.peerConnection = null
+    this.dataChannel = null
+  }
+
+  setLocalViewSrcObject(srcObject) {
+    this.localViewSrcObject = srcObject
+    this.setState({
+      localViewSrcObject: srcObject
+    })
+  }
+
+  setRemoteViewSrcObject(srcObject) {
+    this.remoteViewSrcObject = srcObject
+    this.setState({
+      remoteViewSrcObject: srcObject
+    })
+  }
+
+  async start(credentials, params) {
+
+    const {
+      region,
+      endpoint,
+      channelName,
+      clientId,
+      sendAudio,
+      sendVideo,
+      openDataChannel,
+      widescreen,
+      forceTURN,
+      natTraversalDisabled,
+      useTrickleICE,
+    } = params
+
+
+    this.webRtcClient = new WebRtcClient(
+      Role.MASTER,
+      credentials,
+      region,
+      endpoint,
+      channelName,
+      clientId,
+      natTraversalDisabled,
+      forceTURN,
+    )
+
+    const channelARN = await this.webRtcClient.getChannelARN()
+    console.log('[MASTER] Channel ARN: ', channelARN)
+
+    const endpointsByProtocol = await this.webRtcClient.getEndpointsByProtocol()
+    console.log('[MASTER] Endpoints: ', endpointsByProtocol)
+
+    const iceServers = await this.webRtcClient.getIceServers()
+    console.log('[MASTER] ICE servers: ', iceServers)
+
+    this.signalingClient = await this.webRtcClient.getSignalingClient()
+
+    const configuration = {
+      iceServers,
+      iceTransportPolicy: forceTURN ? 'relay' : 'all',
+    };
+
+    const resolution = widescreen ? { width: { ideal: 1280 }, height: { ideal: 720 } } : { width: { ideal: 640 }, height: { ideal: 480 } };
+    const constraints = {
+      video: sendVideo ? resolution : false,
+      audio: sendAudio,
+    };
+
+    // Get a stream from the webcam and display it in the local view.
+    // If no video/audio needed, no need to request for the sources.
+    // Otherwise, the browser will throw an error saying that either video or audio has to be enabled.
+    if (sendVideo || sendAudio) {
```

Comment:
To confirm, this means the JS page will allow a `recvonly` option?

Reply:
All of my code is simply a refactoring of the code here: https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-js/tree/master/examples